hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/state/PersistentState.java

```diff
@@ -126,54 +131,51 @@ private void removeDeadDataNodes() {
   }
 
   public List<String> getDeadJournalNodes() {
+    List<String> deadJournalHosts = new ArrayList<>();
+
     if (deadJournalNodeTimeStamp != null && deadJournalNodeTimeStamp.before(new Date())) {
       removeDeadJournalNodes();
-      return new ArrayList<>();
     } else {
       Map<String, String> journalNodes = getJournalNodes();
-      Set<String> journalHosts = journalNodes.keySet();
-      List<String> deadJournalHosts = new ArrayList<>();
-      for (String journalHost : journalHosts) {
-        if (journalNodes.get(journalHost) == null) {
-          deadJournalHosts.add(journalHost);
```

Comment:
I don't think I don't understand.   The optimal way to iterate when needing to access both the key and the value is to iterate over the entry set.   This change was driven specifically by the quality checks.    I don't see how this has anything to do with nested method calls.